src/components/modals/SwapCoins/InputModal/index.tsx

```diff
@@ -0,0 +1,82 @@
+import Button from "components/inputs/Button";
+import InputAmount from "components/inputs/InputAmount";
+import ModalContainer from "components/modals/ModalContainer";
+import React, { useCallback, useState } from "react";
+import { InputModalProps, Swap } from "../types";
+import { ButtonContainer, IconContainer } from "./styles";
+import SwapIcon from "assets/icons/swap-icon.svg";
+import { dark } from "theme";
+import { Gap } from "utils/gap";
+
+const InputModal = ({
+  avaliableCoins,
+  continueButtonOnClick = () => {},
+  onClickClose = () => {},
+  getSwapRate = () => {},
+}: InputModalProps) => {
+  const [swapFrom, setSwapFrom] = useState<Swap>();
+  const [swapTo, setSwapTo] = useState<Swap>();
+  const [intervalId, setIntervalId] = useState<any>();
+  const rate = getSwapRate({ swapFrom: swapFrom, swapTo: swapTo });
+
+  const rateChanged = useCallback(() => {
+    setSwapTo({ value: rate });
+  }, [swapTo]);
+
+  const swapFromChanged = useCallback(
+    ({ token, value }: Swap) => {
+      setSwapFrom({ token: token, value: value });
+    },
+    [swapFrom],
+  );
+
+  const swapToChanged = useCallback(
+    ({ token }: Swap) => {
```

Comment:
Probably better to extract the setInterval in an useEffect: https://upmostly.com/tutorials/setinterval-in-react-components-using-hooks
But I don’t understand what you want to do with the rate